### ISSUE TITLE
Invalid Referer from knight proxy

### ISSUE THREAD

**Issue by yosun** [NUKnightLab/TimelineJS3#835]
Just checked a timeline js site from a few years ago and was unhappy to find that it no longer works?

[![Image from Gyazo](https://i.gyazo.com/424878e4bc17ef1b01f6698df13f7887.png)](https://gyazo.com/424878e4bc17ef1b01f6698df13f7887)

error from proxy https://sheets-proxy.knightlab.com/proxy/

```
{
message: "Invalid referrer",
proxy_err_code: "invalid_for_proxy"
}
```

**Comment by JoeGermuska** [comment]
(to be honest, this would be more appropriate as [a support request](https://knightlab.zendesk.com/hc/en-us/requests/new) than a GitHub issue, but we'll answer here and maybe it will be helpful for someone in the future)

As part of [our response to Google turning off the Google Sheets v3 API](https://knightlab.northwestern.edu/2020/09/16/timelinejs-update-2/), we introduced a proxy service which can read data from Google Sheets. We did not want other projects responding to this change to take advantage of our resources, so the proxy service makes some checks to ensure it is only used for timelines.

The proxy is permissive for timelines served from our own CDN (that is, embeds generated using the tool at timeline.knightlab.com); however, for self-hosted timelines (including using the Wordpress plugin), it requires that the Google Sheet only use the standard column headers.

So, probably, you have added other columns to your spreadsheet. Remove those columns and the timeline should work.

**Comment by yosun** [comment]
Hmm so how do we get this to work with a spreadsheet that is basically 99% based on your template but i added in a tags column? Can I simply self host it?

**Comment by JoeGermuska** [comment]
At this time, TimelineJS can only read tabular data from Google Sheets. The alternative would be to use the JSON format, documented at http://timeline.knightlab.com/docs/json-format.html and for which there are several examples in this repository.

As part of the process of adapting to the changes with the Google Sheets API, TimelineJS does read process the Google Sheets data in CSV format; with some refactoring in `ConfigFactory.js` one could extend the code to read a CSV directly from a URL, instead of reading from Google Sheets, or reading a JSON URL.  This work is not currently on our timeline, but we would gratefully consider a pull request which provides support for this feature.

tl;dr: the easiest way to get it to work is to remove the tags column or to use an iframe embed from Timeline.knightlab.com
